plugins/home/api-report.md

```diff
@@ -66,8 +71,19 @@ export function createCardExtension<T>(options: {
   title: string;
   components: () => Promise<ComponentParts>;
   name?: string;
+  data?: {
+    homepageSettings?: HomepageSettings;
+  };
```

Comment:
Ping @Rugvip here too.

I was thinking maybe something along these lines? I think it'd be good to not have any sort of `data` exposed raw.

```ts
export function createCardExtension<T>(options: {
  title: string;
  components: () => Promise<ComponentParts>;
  name?: string;
  description?: string;
  layout?: {
    width?: { minColumns?: number, maxColumns?: number, defaultColumns?: number };
    height?: { minColumns?: number, maxColumns?: number, defaultColumns?: number };
  };
  settings?: {
    schema?: RJSFSchema;
  };
```

in all honesty I was tempted to move towards grouping "metadata" together as well, so you have `title`, `description`, future `iconUrl` and `homepageUrl` etc all in one place under a `metadata?: {}` object. But that can be done separately together with some judicious use of deprecations to let people move existing usages of `title` etc over to it gradually.

Regarding `description`, is that one perhaps so generically useful that you can put it together with `title` in `CardExtensionProps` and the others above?

Reply:
hm. The width.minColumns etc. does not belong to widget, they are part of the Grid. Widget has these minWidth, maxWidth etc. properties instead.